src/compose/MentionWarnings.js

```diff
@@ -0,0 +1,186 @@
+/* @flow strict-local */
+
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+import type { Auth, Stream, Dispatch, Narrow, UserOrBot, Subscription, GetText } from '../types';
+import { TranslationContext } from '../boot/TranslationProvider';
+import { getActiveUsersById, getAuth } from '../selectors';
+import { isPrivateNarrow } from '../utils/narrow';
+import * as api from '../api';
+import { showToast } from '../utils/info';
+
+import AnimatedScaleComponent from '../animation/AnimatedScaleComponent';
+import MentionedUserNotSubscribed from '../message/MentionedUserNotSubscribed';
+
+type State = {|
+  unsubscribedMentions: Array<number>,
+|};
+
+type SelectorProps = {|
+  auth: Auth,
+  usersById: Map<number, UserOrBot>,
+|};
+
+type Props = $ReadOnly<{|
+  narrow: Narrow,
+  stream: Subscription | {| ...Stream, in_home_view: boolean |},
+
+  dispatch: Dispatch,
+  ...SelectorProps,
+|}>;
+
+class MentionWarnings extends PureComponent<Props, State> {
+  static contextType = TranslationContext;
+  context: GetText;
+
+  state = {
+    unsubscribedMentions: [],
+  };
+
+  /**
+   * Tries to parse a user object from an @-mention.
+   *
+   * @param completion The autocomplete option chosend by the user.
+      See JSDoc for AutoCompleteView for details.
+   */
+  getUserFromMention = (completion: string): UserOrBot | void => {
+    const { usersById } = this.props;
+
+    const unformattedMessage = completion.split('**')[1];
+    const [userFullName, userIdRaw] = unformattedMessage.split('|');
+
+    // We skip user groups, for which autocompletes are of the form
+    // `*<user_group_name>*`, and therefore, message.split('**')[1]
+    // is undefined.
+    if (unformattedMessage === undefined) {
+      return undefined;
+    }
+
+    if (userIdRaw !== undefined) {
+      const userId = Number.parseInt(userIdRaw, 10);
+      return usersById.get(userId);
+    }
+
+    for (const user of usersById.values()) {
+      if (user.full_name === userFullName) {
+        return user;
+      }
+    }
+
+    return undefined;
+  };
+
+  showSubscriptionStatusLoadError = (mentionedUser: UserOrBot) => {
+    const _ = this.context;
+
+    const alertTitle = _.intl.formatMessage(
+      {
+        id: "Couldn't load information about {fullName}",
+        defaultMessage: "Couldn't load information about {fullName}'",
```

Comment:
```suggestion
        id: "Couldn't load information about {fullName}",
        defaultMessage: "Couldn't load information about {fullName}",
```

This also serves as a reminder that we should really make our own little wrapper for `formatMessage` that saves us from having to repeat the string twice :slightly_smiling_face: